src/tests/test_log_dev.cpp

```diff
@@ -13,135 +13,307 @@
+    LOGINFO("Step 1: Create a single logstore to start rollback test");
+    auto logdev_id = logstore_service().create_new_logdev();
+    s_max_flush_multiple = logstore_service().get_logdev(logdev_id)->get_flush_size_multiple();
+    auto log_store = logstore_service().create_new_log_store(logdev_id, false);
+    auto store_id = log_store->get_store_id();
+
+    auto restart = [&]() {
+        std::promise< bool > p;
+        auto starting_cb = [&]() {
+            logstore_service().open_logdev(logdev_id);
+            logstore_service().open_log_store(logdev_id, store_id, false /* append_mode */).thenValue([&](auto store) {
+                log_store = store;
+                p.set_value(true);
+            });
+        };
+        start_homestore(true /* retart*/, starting_cb);
+        p.get_future().get();
+    };
+
+    LOGINFO("Step 2: Issue sequential inserts with q depth of 10");
+    logstore_seq_num_t cur_lsn = 0;
+    kickstart_inserts(log_store, cur_lsn, 500);
+
+    LOGINFO("Step 3: Rollback last 50 entries and validate if pre-rollback entries are intact");
+    rollback_validate(log_store, cur_lsn, 50); // Last entry = 450
+
+    LOGINFO("Step 4: Append 25 entries after rollback is completed");
+    kickstart_inserts(log_store, cur_lsn, 25); // Last entry = 475
+
+    LOGINFO("Step 5: Rollback again for 75 entries even before previous rollback entry");
+    rollback_validate(log_store, cur_lsn, 75); // Last entry = 400
+
+    LOGINFO("Step 6: Append 25 entries after second rollback is completed");
+    kickstart_inserts(log_store, cur_lsn, 25); // Last entry = 425
+
+    LOGINFO("Step 7: Restart homestore and ensure all rollbacks are effectively validated");
+    restart();
+
+    LOGINFO("Step 8: Post recovery, append another 25 entries");
+    kickstart_inserts(log_store, cur_lsn, 25); // Last entry = 450
+
+    LOGINFO("Step 9: Rollback again for 75 entries even before previous rollback entry");
+    rollback_validate(log_store, cur_lsn, 75); // Last entry = 375
+
+    LOGINFO("Step 10: After 3rd rollback, append another 25 entries");
+    kickstart_inserts(log_store, cur_lsn, 25); // Last entry = 400
+
+    LOGINFO("Step 11: Truncate all entries");
+    truncate_validate(log_store);
+
+    LOGINFO("Step 12: Restart homestore and ensure all truncations after rollbacks are effectively validated");
+    restart();
+
+    LOGINFO("Step 13: Append 25 entries after truncation is completed");
+    kickstart_inserts(log_store, cur_lsn, 25); // Last entry = 425
+
+    LOGINFO("Step 14: Do another truncation to effectively truncate previous records");
+    truncate_validate(log_store);
+
+    LOGINFO("Step 15: Validate if there are no rollback records");
+    rollback_records_validate(log_store, 0 /* expected_count */);
+}
+
+SISL_OPTION_GROUP(test_log_dev,
+                  (num_logdevs, "", "num_logdevs", "number of log devs",
+                   ::cxxopts::value< uint32_t >()->default_value("4"), "number"),
```

Comment:
Are we using num_logdevs? Also shouldn't we create multiple logdevs as part of the test (are we planning to do it in later PR?)

Reply:
Yes we are creating multiple logdev 's test_log_store and also in WriteSyncThenRead. Planning to add more test cases in future PR.